linera-chain/src/certificate.rs

```diff
@@ -273,6 +295,10 @@ impl<T> GenericCertificate<T> {
         &self.signatures
     }
 
+    pub fn signatures_mut(&mut self) -> &mut Vec<(ValidatorName, Signature)> {
+        &mut self.signatures
+    }
```

Comment:
This seems to be test-only, and also feels dangerous. Maybe let's use `cfg(with_testing)` here.

Reply:
Done in [Review part 1](https://github.com/linera-io/linera-protocol/pull/2877/commits/e1348d1fc4206a1912eea5f7deb4f771c5d9084f).